src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py

```diff
@@ -140,16 +142,27 @@ def parse_issues(self, clang_output, revision):
         if not headers:
             raise Exception('No clang-tidy header was found even though a clang output was provided')
 
+        def _parse_body(start_pos, end_pos):
+            assert isinstance(start_pos, int)
+            assert isinstance(end_pos, int)
+            body = clang_output[start_pos:end_pos]
+            footer = REGEX_FOOTER.search(body)
+            if footer is None:
+                return body
+            return body[:footer.start()-1]
+
         issues = []
         for i, header in enumerate(headers):
             issue = ClangTidyIssue(header.groups(), revision)
 
             # Get next header
             if i+1 < len(headers):
+                # Parse body until next header
                 next_header = headers[i+1]
-                issue.body = clang_output[header.end():next_header.start() - 1]
+                issue.body = _parse_body(header.end(), next_header.start() - 1)
             else:
-                issue.body = clang_output[header.end():]
+                # Limit last element to 3 lines to avoid parsing extra metadatas
```

Comment:
Are we sure the rest is always extra metadata we are not interested in? Can we assert it somehow?

Reply:
I guess it's hard, I don't see anything that would clearly separate the part we are not interested in...
So we'll have to rely on clang-tidy's output not to change too much. And maybe you could add a test that actually runs clang-tidy instead of mocking its output? This way when we update it we notice if something is broken.

Reply:
Not really doable right now, because we would need to download the clang-tidy from Taskcluster in unit tests... that would be for an integration tests.

I [created an issue](https://github.com/mozilla-releng/services/issues/1255) to track the integration tests to implement, feel free to add some